### PR TITLE
[Fixes] Explore page fixes

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

#180581763

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/SearchSection/index.js [hunk 1]
```diff
@@ -104,6 +104,16 @@ const SearchSection = ({
...
+            <Button
+              variant="outlined"
+              className={classes.saveButton}
+              href="/account/searches"
+              component={Link}
+              underline="none"
+            >
+              Load search
+            </Button>
+
```
Comment (kilemensi) [inline src/components/SearchSection/index.js:116]: Like **Save search**, this button should also only appear if you're logged in.

File: src/components/SearchResults/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    fontSize: "64px",
+    lineHeight: "110%",
+    fontFamily: typography.h1.fontFamily,
+  },
+}));
+
+const SearchResults = ({ results, ...props }) => {
+  const classes = useStyles(props);
+
+  return (
```
Comment (kilemensi) [inline src/components/SearchResults/index.js:21]:
```suggestion

  if (!results?.length) {
    return null;
  }
  return (
```

File: src/components/SearchResults/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <Section className={classes.section}>
+      <Typography variant="body1" className={classes.searchQuery}>
+        Search results: {results}
```
Comment (kilemensi) [inline src/components/SearchResults/index.js:24]: Search results: Should be some label prop

File: src/components/TweetsContainer/index.js [hunk 4]
```diff
@@ -175,6 +176,7 @@ function TweetsContainer({
       />
       {isLoading && <Loading />}
       <Chart {...props} data={insights} classes={{ root: classes.chartRoot }} />
+      {!query ? null : <SearchResults results={query} />}
```
Comment (kilemensi) [inline src/components/TweetsContainer/index.js:179]: Let `SearchResults` check if results is null rather than doing the check here.

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

Where is the screenshot to show changes before/after?

**Review by KobbyMmo — COMMENTED** [review]
From the Figma, the explore page shouldn't show this title
image

**Comment by kilemensi** [comment]
> From the Figma, the explore page shouldn't show this title

I think 👇🏽 is what this PR is trying to do @KobbyMmo... Rather than creating yet a different page, we're reusing the same explore page.
Screenshot from 2021-12-14 18-45-28

**Comment by KobbyMmo** [comment]
> I think 👇🏽 is what this PR is trying to do @KobbyMmo... Rather than creating yet a different page, we're reusing the same explore page.

Then I think it will be better to have the title section as sticky. @kilemensi

**Comment by kilemensi** [comment]
> > I think 👇🏽 is what this PR is trying to do @KobbyMmo... Rather than creating yet a different page, we're reusing the same explore page.
> 
> Then I think it will be better to have the title section as sticky. @kilemensi Or maybe in another PR

Lets make it work first @KobbyMmo

File: src/components/SearchResults/index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  },
+}));
+
+const SearchResults = ({ results, label, ...props }) => {
+  const classes = useStyles(props);
+
+  if (!results?.length) {
+    return null;
+  }
+
```
Comment (kilemensi) [inline src/components/SearchResults/index.js:26]:
```suggestion
```

File: src/components/SearchSection/index.js [hunk 6]
```diff
@@ -104,6 +104,18 @@ const SearchSection = ({
...
+                className={classes.saveButton}
+                href="/account/searches"
+                component={Link}
+                underline="none"
+              >
+                Load search
+              </Button>
+            ) : null}
+
             {session ? (
```
Comment (kilemensi) [inline src/components/SearchSection/index.js:119]: A single `{session ?` check should be enough to add both **Load search** and **Save search** buttons

File: src/components/TweetsContainer/index.js [hunk 7]
```diff
@@ -175,6 +176,7 @@ function TweetsContainer({
       />
       {isLoading && <Loading />}
       <Chart {...props} data={insights} classes={{ root: classes.chartRoot }} />
+      <SearchResults results={query} label="Search Results" />
```
Comment (kilemensi) [inline src/components/TweetsContainer/index.js:179]:
```suggestion
      <SearchResults query={query} label="Search Results" />
```
I think `query` or `keyword` is better than `results` here.

**Review by kilemensi — APPROVED** [review]
🚀

File: src/components/SearchResults/index.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    padding: `${typography.pxToRem(48)} ${typography.pxToRem(0)}`,
+  },
+  searchQuery: {
+    fontStyle: "normal",
+    fontWeight: "bold",
+    fontSize: typography.pxToRem(64),
+    lineHeight: "110%",
+    fontFamily: typography.h1.fontFamily,
+  },
+}));
```
Comment (KhadijaMahanga) [inline src/components/SearchResults/index.js:18]: @nerdlet this is `h2`

File: src/components/SearchResults/index.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+  if (!results?.length) {
+    return null;
+  }
+
+  return (
+    <Section className={classes.section}>
+      <Typography variant="body1" className={classes.searchQuery}>
+        {label} : {results}
+      </Typography>
```
Comment (KhadijaMahanga) [inline src/components/SearchResults/index.js:31]:
```suggestion
      <Typography variant="h2">
        {label} : {results}
      </Typography>
```

**Review by KhadijaMahanga — APPROVED** [review]
🚀 🚀

**Pull request merged**